### DISCUSSION TITLE
How to add someone to a list via API

### DISCUSSION THREAD

**Discussion by lassehjorthmadsen** [bluesky-social/atproto#2084] [How to do it?]
I'm new with the atproto API, so bear with me:
I'm trying to write a piece of R code that adds someone to a list on my account. For example, I can follow a user programmatically using the below code. What do I need to do to add a user did to a list (and perhaps create a list first)? Couldn't find it anywhere.
Thanks a lot.

```
follow_record <- list(
  "$type" = "app.bsky.graph.follow",
  "createdAt" = format(Sys.time(), "%Y-%m-%dT%H:%M:%SZ"),
  "subject" = follow_did
)

follow_data <- list(
  repo = my_did,
  collection = "app.bsky.graph.follow",
  record = follow_record
)

req_follow <-
  httr2::request("https://bsky.social/xrpc/com.atproto.repo.createRecord") %>% 
  httr2::req_body_json(data = follow_data) %>%
  httr2::req_auth_bearer_token(token = token) %>% 
  httr2::req_perform() 

```

**Comment by bnewbold** [comment]
Hi @lassehjorthmadsen, totally reasonable question! We are actually working on some docs for lists right now and they should be ready in the next week or two.

@pfrazee any chance you could copy/paste some work-in-progress here?
